src/PanelTraits/Update.php

```diff
@@ -73,4 +75,31 @@ public function getUpdateFields($id)
 
         return $fields;
     }
+
+    /**
+     * Get the value of the 'name' attribute from the declared relation model in the given field.
+     *
+     * @param Model $model The current CRUD model.
+     * @param array $field The CRUD field array.
+     *
+     * @return mixed The value of the 'name' attribute from the relation model.
+     */
+    private function getModelAttributeValue($model, $field)
+    {
+        if (isset($field['entity'])) {
+            $relationArray = explode('.', $field['entity']);
+            $relatedModel = array_reduce(array_splice($relationArray, 0, -1), function ($obj, $method) {
+                return $obj->{$method} ? $obj->{$method} : $obj;
+            }, $model);
+
+            $relationMethod = end($relationArray);
+            if ($relatedModel->{$relationMethod} && $relatedModel->{$relationMethod}() instanceof HasOneOrMany) {
```

Comment:
Could we turn this into a ternary to make it more confusing? 👿 

Why only `HasOneOrMany` and not any of the other types?

Reply:
`$relatedModel->{$relationMethod}` is a dynamic method to the related record instance. For example, $user->phone will retrieve an associated phone object, or null if none defined.
`$relatedModel->{$relationMethod}()` is the actual Eloquent relation definition, which is always non-null (if it actually defined on the model, of course).

So, what I'm checking here first is if we have a non-null property on the model. If that's the case, that can either be a related record (because of Eloquent's dynamic properties which "allow you to access relationship methods as if they were properties defined on the model") or an actual property on the model. If it's the former, we have to get the attribute from the related model.

Reply:
And yes, I know it's ugly as all hell. If you have a better suggestion as to which we can make this less confusing, it would be greatly appreciated.

Reply:
Extract `array_splice($relationArray, 0, -1)` into some sensible method, private.

Humour me and change this:

```
          $relatedModel = array_reduce(array_splice($relationArray, 0, -1), function ($obj, $method) {
+                return $obj->{$method} ? $obj->{$method} : $obj;
+            }, $model);
```

...into procedural code. Functional is **not** helping out here.

Extract `return $obj->{$method} ? $obj->{$method} : $obj;` into a sensible method, private.

And get rid of the ternary.

And comment it.

And it'll read better; it'll be much longer but atm I can't really figure out what it does.